pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -166,137 +204,153 @@ func (computeCfg *computeServiceConfig) getManagedVnets() map[string]armnetwork.
-	if err != nil {
-		azurePluginLogger().Error(err, "failed to fetch cloud resources", "account", computeCfg.account)
-		return err
-	} else {
-		exists := struct{}{}
-		vnetIDs := make(map[string]struct{})
-		vpcPeers := computeCfg.buildMapVpcPeers(vnets)
-		vmIDToInfoMap := make(map[internal.InstanceID]*virtualMachineTable)
+	allVirtualMachines := make(map[types.NamespacedName][]*virtualMachineTable)
+	vnetIDs := make(map[string]struct{})
+	// For cases where polling is done for only changed selectors, update snapshot only for these selectors without disturbing others.
+	// TODO: there is a bug in this case, if a vpc becomes unmanaged, vnetIDs would still show it as managed until next poll interval.
+	if lenSelectorConfigs != len(computeCfg.selectorConfigMap) {
+		snapshot := computeCfg.resourcesCache.GetSnapshot()
+		if snapshot != nil {
+			// Load the present snapshot details.
+			allVirtualMachines = snapshot.(*cacheSnapshot).vms
+			vnetIDs = snapshot.(*cacheSnapshot).vnetIDs
+		}
+	}
+	for _, selectorObject := range selectorConfigs {
+		virtualMachines, err := computeCfg.getVirtualMachines(selectorObject)
+		if err != nil {
+			azurePluginLogger().Error(err, "failed to fetch cloud resources", "account", computeCfg.accountNamespacedName)
+			return err
+		}
 		for _, vm := range virtualMachines {
-			id := internal.InstanceID(strings.ToLower(*vm.ID))
-			vmIDToInfoMap[id] = vm
-			vnetIDs[*vm.VnetID] = exists
+			vnetIDs[*vm.VnetID] = struct{}{}
 		}
-		computeCfg.resourcesCache.UpdateSnapshot(&computeResourcesCacheSnapshot{vmIDToInfoMap, vnets, vnetIDs, vpcPeers})
+		allVirtualMachines[selectorObject.namespacedName] = virtualMachines
 	}
+	computeCfg.resourcesCache.UpdateSnapshot(&cacheSnapshot{allVirtualMachines, vnets, vnetIDs, vnetPeers})
+
 	return nil
 }
 
 func (computeCfg *computeServiceConfig) AddResourceFilters(selector *crdv1alpha1.CloudEntitySelector) error {
 	subscriptionIDs := []string{computeCfg.credentials.SubscriptionID}
 	tenantIDs := []string{computeCfg.credentials.TenantID}
 	locations := []string{computeCfg.credentials.region}
-
+	namespacedName := types.NamespacedName{Namespace: selector.Namespace, Name: selector.Name}
 	if filters, ok := convertSelectorToComputeQuery(selector, subscriptionIDs, tenantIDs, locations); ok {
-		key := selector.GetNamespace() + "/" + selector.GetName()
-		computeCfg.computeFilters[key] = filters
-		computeCfg.selectors[key] = selector.DeepCopy()
+		azurePluginLogger().Info("resource query filters created", "selector", namespacedName)
+		computeCfg.configChangedSelectors = append(computeCfg.configChangedSelectors, &namespacedName)
+		computeCfg.selectorConfigMap[namespacedName] = &selectorConfig{
+			namespacedName:  namespacedName,
+			instanceFilters: filters,
+			ces:             selector.DeepCopy(),
+		}
 	} else {
-		return fmt.Errorf("error creating resource query filters")
+		return fmt.Errorf("error creating resource query filters for selector: %v", namespacedName)
```

Comment:
No need to add selector info the error msg, I believe the final consumer is adding the selector namespacedname. Please check.

Reply:
yes, selector is added to the message at the account manager level, hence removed from here